sechub-wrapper-owasp-zap/src/main/java/com/mercedesbenz/sechub/owaspzapwrapper/scan/AbstractScan.java

```diff
@@ -289,8 +298,40 @@ protected void setupAdditonalProxyConfiguration() throws ClientApiException {
         }
     }
 
+    protected void deactivateRules() throws ClientApiException {
+        OwaspZapFullRuleset fullRuleset = scanConfig.getFullRuleset();
+        DeactivatedRuleReferences deactivatedRuleReferences = scanConfig.getDeactivatedRuleReferences();
+        if (fullRuleset == null && deactivatedRuleReferences == null) {
+            return;
+        }
+        List<RuleReference> rulesReferences = deactivatedRuleReferences.getDeactivatedRuleReferences();
+        if (rulesReferences == null) {
+            return;
+        }
+
+        for (RuleReference ruleRef : rulesReferences) {
+            Rule ruleToDeactivate = fullRuleset.findRuleByRef(ruleRef.getRef());
+
+            if (isPassiveRule(ruleToDeactivate.getType())) {
+                clientApi.pscan.disableScanners(ruleToDeactivate.getId());
+            } else if (isActiveRule(ruleToDeactivate.getType())) {
+                // null specifies the default policy
+                clientApi.ascan.disableScanners(ruleToDeactivate.getId(), null);
+            }
+        }
+    }
+
+    private boolean isPassiveRule(String type) {
+        return "passive".equals(type);
```

Comment:
do a type to lower case. Some user could provide a rule type of "PaSsive" or so.